src/hb-ot-color.cc

```diff
@@ -146,38 +158,58 @@ hb_ot_color_get_palette_flags (hb_face_t *face, unsigned int palette)
  * Since: REPLACEME
  */
 unsigned int
-hb_ot_color_get_palette_colors (hb_face_t       *face,
-				unsigned int     palette, /* default=0 */
-				unsigned int     start_offset,
-				unsigned int    *color_count /* IN/OUT */,
-				hb_ot_color_t   *colors /* OUT */)
+hb_ot_color_get_palette_colors (hb_face_t      *face,
+				unsigned int    palette, /* default=0 */
+				unsigned int    start_offset,
+				unsigned int   *count /* IN/OUT */,
+				hb_ot_color_t  *colors /* OUT */)
 {
   const OT::CPAL& cpal = _get_cpal(face);
-  if (unlikely (palette >= cpal.numPalettes))
```

Comment:
This work is great, thanks!

I believe you can move the logic inside this and `hb_ot_color_get_color_layers` to the table itself and use this as a proxy only.